### DISCUSSION TITLE
First connexion : the "next" button is not available

### DISCUSSION THREAD

**Discussion by jmarchand52** [nilsteampassnet/TeamPass#3921] [General]
Hi,
On a fresh Debian 12/PHP 8.2/MariaDB VM, I did the differents tasks for installation.

But when I launch the first access to http://teampass.ijm.fr, I did not get the "Next" button available.

I put Apache into "info" loglevel and I configure PHP to log every error : nothing appears.

Any idea ?

Best regards,

  Joel
![Capture d’écran du 2023-11-01 15-09-04](https://github.com/nilsteampassnet/TeamPass/assets/16597232/0656e4f9-aebf-47bf-8087-2e0e0e123cfe)

**Comment by nilsteampassnet** [comment]
@jmarchand52 
Just click "start" button.
Once all checks done, you will have the button "next" enabled ;)

**Comment by jmarchand52** [comment]
I get a red message at the right of the screen.
![Capture d’écran du 2023-11-01 15-16-00](https://github.com/nilsteampassnet/TeamPass/assets/16597232/a34da3e8-ebd9-4d97-b4e0-56a9b77c8020)

**Comment by jmarchand52** [comment]
And I get into /var/log/apache2/error.log :

PHP Fatal error:  Uncaught Error: Undefined constant "DB_PASSWD" in /var/www/html/sources/main.functions.php:4357\nStack trace:\n#0 /var/www/html/sources/main.functions.php(59): loadClasses()\n#1 /var/www/html/install/install.queries.php(114): require_once('...')\n#2 {main}\n  thrown in /var/www/html/sources/main.functions.php on line 4357, referer: http://teampass.ijm.fr/install/install.php
[Wed Nov 01 15:15:52.891083 2023] [php:error] [pid 729] [client 10.0.2.14:58412] PHP Fatal error:  Uncaught Error: Undefined constant "DB_PASSWD" in /var/www/html/sources/main.functions.php:4357\nStack trace:\n#0 /var/www/html/sources/main.functions.php(59): loadClasses()\n#1 /var/www/html/install/install.queries.php(114): require_once('...')\n#2 {main}\n  thrown in /var/www/html/sources/main.functions.php on line 4357, referer: http://teampass.ijm.fr/install/install.php

**Reply by nilsteampassnet** [reply]
I just give a try and didn't reproduce.
What minor version?
![image](https://github.com/nilsteampassnet/TeamPass/assets/1197546/aa09ab43-f851-4c94-b4b9-d0efbd40938d)
Please get the latest commit.

**Comment by jmarchand52** [comment]
I did this action :
 git clone https://github.com/nilsteampassnet/TeamPass.git

If I launch Firefox into the VM and access to http://localhost/ , I get the same thing.

**Comment by jmarchand52** [comment]
On Debian 11 / PHP 7.4, no the same : look at the screenshot
![Capture d’écran du 2023-11-01 17-06-33](https://github.com/nilsteampassnet/TeamPass/assets/16597232/e5f2c654-41d9-44cb-8978-5d417907255f)

**Reply by nilsteampassnet** [reply]
@jmarchand52 
Can you share in the same time the server error from log file please?

**Comment by MadalinPo** [comment]
Got the same problem, from what I've seen the faulty line was pushed on master yesterday. I've tried multiple times and still got the same problem

**Comment by jmarchand52** [comment]
No line into the Apache logfiles with Debian 11. The wheel turnes several minutes.

**Reply by nilsteampassnet** [reply]
@jmarchand52 
HUm ... can you open the browser console and reperform the operation while checking any error in this console.
If nothing happens at this moment of the installation then an error must occurred somewhere.

**Reply by anastas76** [reply]
Hi! This error present in the last minor version.
apache's error.log:
[proxy_fcgi:error] [pid 15800] [client 192.168.71.1:60197] AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught Error: Undefined constant "DB_PASSWD" in /var/www/html/sources/main.functions.php:4350\nStack trace:\n#0 /var/www/html/sources/main.functions.php(60): loadClasses()\n#1 /var/www/html/install/install.queries.php(114): require_once('...')\n#2 {main}\n  thrown in /var/www/html/sources/main.functions.php on line 4350', referer: https://192.168.71.180/install/install.php

I commented next string in the sources/main.functions.php:

/* if (defined('DB_PASSWD_CLEAR') === false {
define ('DB_PASSWD_CLEAR', defuseReturnDecrypted(DB_PASSWD, []));
*/

And could install TeamPass.

**Comment by Arash9703** [comment]
Got through the same pbm . 
Your www-data ( install.php) user doesn't have access permission to /teampass.

chown -R www-data:www-data /var/www/html/teampass
		
Or # chown -R www-data /var/www/html
& systemctl restart apache2

Normally it should work

**Comment by jmarchand52** [comment]
The two propositions of Anast76 and Arash9703 do not work for me.
I have no message in Apache logs, when the wheel "Working on it" runs.

**Comment by jmarchand52** [comment]
Today, after a new "git clone", I get the Apache2 error :+1:

**Comment by jmarchand52** [comment]
[Sat Nov 11 21:44:50.346388 2023] [php7:error] [pid 618] [client 10.0.2.6:46218] PHP Parse error:  syntax error, unexpected '|', expecting '{' in /var/www/html/sources/main.functions.php on line 2519, referer: http://teampass.ijm.fr/install/install.php

**Comment by arupdeb381** [comment]
Hi, 
can anyone kindly help me on this. during fresh teampass installation on server windows 2019 and wamp 3.3.7 - stuck on this page..( I have tried 3.1.3.9 and 3.1.3.10 both the version)

![image](https://github.com/user-attachments/assets/38a32a2a-7fad-43fd-b8bf-4e1fa1adb957)

error :
[21-Jan-2025 07:10:54 UTC] PHP Fatal error:  Uncaught mysqli_sql_exception: Duplicate key name 'teampass_log_items_id_item_IDX' in C:\wamp64\www\TeamPass-master\install\install.queries.php:607
Stack trace:
#0 C:\wamp64\www\TeamPass-master\install\install.queries.php(607): mysqli_query(Object(mysqli), 'CREATE INDEX te...')
#1 {main}
  thrown in C:\wamp64\www\TeamPass-master\install\install.queries.php on line 607

**Reply by corentin-soriano** [reply]
I think your database was not completely empty during installation.
Can you do `DROP DATABASE`, `CREATE DATABASE` and then relaunch installation?

It is possible to remove the problematic index without dropping database but you will probably get other errors:
`ALTER TABLE teampass_log_items DROP INDEX teampass_log_items_id_item_IDX;`

**Reply by arupdeb381** [reply]
Sorry no luck tried both the option ..still getting the same error.

![image](https://github.com/user-attachments/assets/bd71b1de-e55d-4e83-91ed-be0d3d01a5d8)
